File: backend/app/schemas/restaurant.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class RestaurantBase(BaseModel):
    name: str
    address: str | None = None
    phone: str | None = None


class RestaurantCreate(RestaurantBase):
    pass


class RestaurantUpdate(BaseModel):
    name: str | None = None
    address: str | None = None
    phone: str | None = None


class RestaurantResponse(RestaurantBase):
    id: int
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/types/restaurant.ts
export interface Restaurant {
  id: number;
  name: string;
  address: string;
  phone: string;
}

export interface RestaurantCreate {
  name: string;
  address: string;
  phone: string;
}

export interface RestaurantUpdate {
  name?: string;
  address?: string;
  phone?: string;
}